src/shared/settingsConfiguration.ts

```diff
@@ -56,4 +56,36 @@ export class DefaultSettingsConfiguration implements SettingsConfiguration {
             return false
         }
     }
+    /**
+     * Adds prompt to list of prompts that have been selected not to display again.
+     * @param promptName Name of prompt to append to list
+     */
+    public disable(promptName: string): void {
+        try {
+            let prompts = this.readSetting<string[]>('doNotShowPrompts', [])
+            if (typeof prompts !== 'object' || !Array.isArray(prompts)) {
+                getLogger().warn('setting "doNotShowPrompts" has an unexpected type. Overwriting.')
+                prompts = []
+            }
+            if (!prompts.includes(promptName)) {
+                prompts.push(promptName)
+                this.writeSetting('doNotShowPrompts', prompts, vscode.ConfigurationTarget.Global)
+            }
+        } catch (e) {
+            getLogger().error('Failed to update setting: doNotShowPrompts', e)
+        }
+    }
+    /**
+     * Verifies if a prompt should be displayed again.
+     * @param promptName Name of the prompt
+     * @returns False when prompt was disabled previously
+     */
+    public shouldDisplayPrompt(promptName: string): boolean {
```

Comment:
where do these actually get called?

Reply:
Once this is merged, they will begin to be used in #1908,.

Reply:
it probably makes sense to also convert existing suppress-prompt calls in this PR, there are not too many cases. For example 

https://github.com/aws/aws-toolkit-vscode/blob/e8af78cc02735411718ef982f364c1df3c473228/src/apprunner/wizards/deploymentButton.ts#L37

and

https://github.com/aws/aws-toolkit-vscode/blob/e8af78cc02735411718ef982f364c1df3c473228/src/shared/sam/activation.ts#L346

Reply:
These have been added to this PR.